### PR TITLE
Remove python-future compatibility code

### PR DESCRIPTION
**Pull request by sbesson**
See https://github.com/ome/omero-py/pull/390 for the rationale

The list of changes should be identical i.e.:
- all `future` or `past` import are replaced by Python 3 calls
- all unnecessary `builtins` imports are removed 
- all Python 2 compatibility logic are removed
- the unmaintained Dockerfile is removed

### CODE REVIEW

**Comment by jburel** [comment]
The docker file has been removed. This could be updated. This has been done is other repository e.g. openmicroscopy

**Comment by sbesson** [comment]
> The docker file has been removed. This could be updated. This has been done is other repository e.g. openmicroscopy

I don't mind either way but unlike `openmicroscopy` I don't think this file was consumed by anything at least in terms of CI. Do you have recommendations guidelines on the target operating system or what the Docker image should be doing?

**Comment by jburel** [comment]
i am certainly not using it. If anyone using it for testing purpose, we can upgrade otherwise I won't bother
@joshmoore are you using the dockerfile?

**Comment by jburel** [comment]
In that case, let's remove it.

File: .github/workflows/publish_pypi.yml [hunk 1]
```diff
@@ -7,8 +7,8 @@ jobs:
     name: Build and publish Python distribution to PyPI
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-python@v4
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
```
Comment (jburel) [inline .github/workflows/publish_pypi.yml:11]: The Python version should be specified to avoid warning and potential surprise

**Comment by sbesson** [comment]
Is this in scope for the upcoming OMERO 5.6.11 release?

**Pull request merged**